engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -74,6 +75,32 @@ def scc_activate_product(product, auth, token, email)
       scc_request = prepare_scc_request(uri.path, product, auth, token, email)
       http.request(scc_request)
     end
+
+    def parse_error(error_message, token = nil, email = nil)
+      error_message = error_message[0..(error_message.index('json') - 1)].strip
+      error_message = error_message.gsub(token, '').squish if token
+      error_message = error_message.gsub(email, '').strip if email
+      error_message
+    end
+
+    def scc_check_subscription_expiration(headers, login, logger)
+      auth = headers['HTTP_AUTHORIZATION'] if headers.include?('HTTP_AUTHORIZATION')
+      uri = URI.parse(SYSTEMS_SUBSCRIPTIONS_URL)
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      scc_request = Net::HTTP::Get.new(uri.path, headers(auth))
+      response = http.request(scc_request)
+      unless response.code_type == Net::HTTPOK
+        logger.info "Could not get the system (#{login}) subscription, error: #{response.message} #{response.code}"
+        response.message = SccProxy.parse_error(response.message) if response.message.include? 'json'
+        return { is_active: false, message: response.message }
+      end
+      parsed_response = JSON.parse(response.body)[0]
```

Comment:
You're always checking the first returned subscription here, that won't work in case the system is using multiple subscriptions. For example when the system uses one subscription for SLES and another subscription for and extension like HA.

Reply:
Fixed